components/ui/swiperAction/useSwiperAction.ts

```
@@ -0,0 +1,177 @@
+  // biome-ignore lint/correctness/useExhaustiveDependencies: 첫 마운트시에만 계산
+  useEffect(() => {
+    const updateLayout = () => {
+      const root = getComputedStyle(document.documentElement);
+      const padding = Number(root.getPropertyValue("--layout-padding-x").replace("px", ""));
+
+      containerWidthRef.current = containerRef.current?.clientWidth ?? 0;
+
+      const contentWidth = containerWidthRef.current - 2 * padding;
+      const sw = sidePeekRatio ? contentWidth / (1 + 2 * sidePeekRatio) : contentWidth;
+      elementWidthRef.current = sw;
+      setSlideWidth(sw);
```

ESLint

[error] Error: Cannot access variable before it is declared

`setSlideWidth` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/components/ui/swiperAction/useSwiperAction.ts:44:7
  42 |       const sw = sidePeekRatio ? contentWidth / (1 + 2 * sidePeekRatio) : contentWidth;
  43 |       elementWidthRef.current = sw;
> 44 |       setSlideWidth(sw);
     |       ^^^^^^^^^^^^^ `setSlideWidth` accessed before it is declared
  45 |
  46 |       threshold.current = Math.floor((containerWidthRef.current - 2 * padding) / STANDARD);
  47 |       x.set(calculateLocation(0));

/home/jailuser/git/components/ui/swiperAction/useSwiperAction.ts:57:22
  55 |   const [currentIndex, setCurrentIndex] = useState(0);
  56 |   const [styleGap, setStyleGap] = useState(0);
> 57 |   const [slideWidth, setSlideWidth] = useState<number | null>(null);
     |                      ^^^^^^^^^^^^^ `setSlideWidth` is declared here
  58 |
  59 |   const calculateLocation = (index: number) => {
  60 |     const slice = sidePeekRatio ? elementWidthRef.current * sidePeekRatio : 0;

(react-hooks/immutability)